From 8f746223d0dd7fa54642898855870d4118e3da2f Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Fri, 16 Mar 2012 10:28:26 +0800
Subject: [PATCH 255/635] ENGR00175965 when swith fb mode, there is no audio

Revert "ENGR00175835 clear secondary display instead blank it."

This reverts commit 0624a356297fc9769f9343cb452579c459117e18.

kernel fix the blank/unblank issue. kernel commit id:
92559bcb2658999f2211dcbe65893c86fa17654f

Signed-off-by: b02247 <b02247@freescale.com>
---
 mx6/libgralloc_wrapper/framebuffer.cpp | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/mx6/libgralloc_wrapper/framebuffer.cpp b/mx6/libgralloc_wrapper/framebuffer.cpp
index 8e5c196..01e20fb 100755
--- a/mx6/libgralloc_wrapper/framebuffer.cpp
+++ b/mx6/libgralloc_wrapper/framebuffer.cpp
@@ -292,6 +292,13 @@ static int mapFrameBufferWithParamLocked(struct private_module_t* module, struct
         return -errno;
     }
 
+    if(param != NULL) {
+        int blank = FB_BLANK_UNBLANK;
+        if(ioctl(fd, FBIOBLANK, blank) < 0) {
+            LOGE("<%s, %d> ioctl FBIOBLANK failed", __FUNCTION__, __LINE__);
+        }
+    }
+
     struct fb_fix_screeninfo finfo;
     if (ioctl(fd, FBIOGET_FSCREENINFO, &finfo) == -1) {
         LOGE("<%s,%d> FBIOGET_FSCREENINFO failed", __FUNCTION__, __LINE__);
@@ -528,14 +535,25 @@ static int unMapFrameBuffer(fb_context_t* ctx, struct private_module_t* module)
 {
     int err = 0;
     pthread_mutex_lock(&module->lock);
+
     size_t fbSize = module->framebuffer->size;
     int fd = module->framebuffer->fd;
     void* addr = (void*)(module->framebuffer->base);
-    memset(addr, 0, fbSize);
     munmap(addr, fbSize);
     delete (module->framebuffer);
     module->framebuffer = NULL;
 
+    int blank = 1;
+    if(ioctl(fd, FBIOBLANK, blank) < 0) {
+        LOGE("<%s, %d> ioctl FBIOBLANK failed", __FUNCTION__, __LINE__);
+    }
+
+    if(ctx->mainDisp_fd > 0) {
+        blank = FB_BLANK_UNBLANK;
+        if(ioctl(ctx->mainDisp_fd, FBIOBLANK, blank) < 0) {
+            LOGE("<%s, %d> ioctl FBIOBLANK failed", __FUNCTION__, __LINE__);
+        }
+    }
     close(fd);
     pthread_mutex_unlock(&module->lock);
 
-- 
1.8.0

